File: ISA-project/src/main/java/com/isa/ISAproject/model/CottageRevision.java
package com.isa.ISAproject.model;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

@Entity
public class CottageRevision {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@ManyToOne(fetch = FetchType.LAZY)
	private CottageReservation cottageReservation;
	
	@OneToOne
	private Revision revision;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public CottageReservation getCottageReservation() {
		return cottageReservation;
	}

	public void setCottageReservation(CottageReservation cottageReservation) {
		this.cottageReservation = cottageReservation;
	}

	public Revision getRevision() {
		return revision;
	}

	public void setRevision(Revision revision) {
		this.revision = revision;
	}

	public CottageRevision(Long id, CottageReservation cottageReservation, Revision revision) {
		super();
		this.id = id;
		this.cottageReservation = cottageReservation;
		this.revision = revision;
	}

	public CottageRevision() {
		super();
	}
}

File: frontend/src/app/model/cottageRevision.ts
import { CottageReservation } from "./cottageReservation";
import { Revision } from "./revision";

export interface CottageRevisionInterface {
    id?: number;
    cottageReservation: CottageReservation;
    revision: Revision;
}
export class CottageRevision implements CottageRevisionInterface {
    id?: number;
    cottageReservation: CottageReservation;
    revision: Revision;

    constructor(obj: CottageRevisionInterface) {
        this.id = obj.id;
        this.cottageReservation = obj.cottageReservation;
        this.revision = obj.revision;

    }

}
